nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/cosmos/document/AbstractCosmosDBClientService.java

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.services.azure.cosmos.document;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosException;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.processors.azure.cosmos.document.AzureCosmosDBUtils;
+import org.apache.nifi.services.azure.cosmos.AzureCosmosDBConnectionService;
+
+@Tags({"azure", "cosmos", "document", "service"})
+@CapabilityDescription(
+        "Provides a controller service that configures a connection to Cosmos DB (Core SQL API) " +
+                " and provides access to that connection to other Cosmos DB-related components."
+)
+public abstract class AbstractCosmosDBClientService
+        extends AbstractControllerService
+        implements AzureCosmosDBConnectionService {
+
+    protected CosmosClient cosmosClient;
+
+    @OnStopped
+    public final void onStopped() {
+        if (this.cosmosClient != null) {
+            try {
+                cosmosClient.close();
+            } catch(CosmosException e) {
+                getLogger().error("Closing CosmosClient Failed: " + e.getMessage(), e);
+            } finally {
+                this.cosmosClient = null;
+            }
+        }
+    }
+
+    protected void createCosmosClient(final String uri, final String accessKey, final String selectedConsistency){
+        final ConsistencyLevel cLevel;
+
+        switch(selectedConsistency) {
+            case AzureCosmosDBUtils.CONSISTENCY_STRONG:
+                cLevel =  ConsistencyLevel.STRONG;
+                break;
+            case AzureCosmosDBUtils.CONSISTENCY_CONSISTENT_PREFIX:
+                cLevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                break;
+            case AzureCosmosDBUtils.CONSISTENCY_BOUNDED_STALENESS:
+                cLevel = ConsistencyLevel.BOUNDED_STALENESS;
+                break;
+            case AzureCosmosDBUtils.CONSISTENCY_EVENTUAL:
+                cLevel = ConsistencyLevel.EVENTUAL;
+                break;
+            default:
+                cLevel = ConsistencyLevel.SESSION;
+        }
+        this.cosmosClient = new CosmosClientBuilder()
+                .endpoint(uri)
+                .key(accessKey)
+                .consistencyLevel(cLevel)
+                .buildClient();
+    }
+
+    @Override
+    public CosmosClient getCosmosClient() {
+        return this.cosmosClient;
+    }
+
+    public void setCosmosClient(CosmosClient client) {
```

Comment:
Is this method used only for testing?  If so, it would be better to override the `cosmosClient` directly instead of having a public set method.

Reply:
This code was already there, it was just moved to another file in current PR.
I am ok to update, can you suggest an example how do you want me to update this.

Reply:
Thanks for the reply and noting the existing usage.  As a simple change, making the `cosmosClient` member variable `protected` would allow assigning it in `MockTestBase`, then this method could be removed.